build/builtin_actors_gen_test.go

```diff
@@ -0,0 +1,59 @@
+package build_test
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"testing"
+
+	"github.com/ipfs/go-cid"
+	"github.com/ipld/go-car/v2"
+	"github.com/stretchr/testify/require"
+
+	"github.com/filecoin-project/lotus/build"
+)
+
+func TestEmbeddedBuiltinActorsMetadata(t *testing.T) {
+	for _, subject := range build.EmbeddedBuiltinActorsMetadata {
+		if subject.BundleGitTag == "" {
+			// Only the latest network version specifies BundleGitTag.
+			// Move on to the next Metadata instead of skipping in order to reduce test log verbosity.
+			continue
+		}
+		subject := subject
+		t.Run(fmt.Sprintf("%s/%v", subject.Network, subject.Version), func(t *testing.T) {
+			carURL := fmt.Sprintf("https://github.com/filecoin-project/builtin-actors/releases/download/%s/builtin-actors-%s.car", subject.BundleGitTag, subject.Network)
```

Comment:
We already have this car file in `build/actors`.

- If we're trying to ensure that our bundle is correct, I'd consider downloading the sha256 file instead and checking that against our local bundle.
- Otherwise, I assume running "make gen" and checking the diff _should_ validate that everything is correct.

I'd just like to avoid having to constantly re-download bundles in CI. Unless there's a reason for this that I'm missing.

Reply:
Ok, you know what? That's ~10s even on my terrible connection. There's little reason not to merge this unless it shows up as actually _costing_ something.